libcontainer/cgroups/stats.go

```diff
@@ -32,9 +32,22 @@ type CpuUsage struct {
 	UsageInUsermode uint64 `json:"usage_in_usermode"`
 }
 
+type PsiData struct {
+	Avg10  float64 `json:"avg10"`
+	Avg60  float64 `json:"avg60"`
+	Avg300 float64 `json:"avg300"`
```

Comment:
Maybe: `Avg map[int]float64` (key=10,60,300)

Reply:
i don't like that very much because the key avg10,60,300 is fixed, so invidual fields are much better for discovery.

Reply:
I was thinking about it, too. If this is fixed in the same way as, say, loadavg is fixed for 1, 5, 15 minutes, we can just use an array with 3 numbers.

OTOH maybe the best way is to have
```
Some, Full map [string]float64
```
where map keys are `avgXXX` and `total`.

In case we want `total` as an integer, maybe it's better to use whatever @AkihiroSuda is proposing, plus a separate Total field.

Reply:
the drawbacks of array or map is non-discoverability of keys and extra allocation where it's not necessary. I don't yet see anything that is bad with explicit keys.

We can make the same arguments with a lot of other statistics too such as io.stat. I think explicit struct are much better to use.